### PR TITLE
Refactor notifier to use Bot interface to send notifications

### LINKED ISSUE #270: [Refactor] Refactor notifier to use Bot interface to send events
**Is your feature request related to a problem? Please describe.**

In notifier, we send events by creating new connection to communication medium on every new event. We can reduce number of connections by sending messages via Bot interface. 

**Describe the solution you'd like**

Implement required methods in Bot interface so that it satisfies Notifier interface. Then add Bot object as a notifier in main function

### PR DESCRIPTION
**Pull request by PrasadG193**
##### ISSUE TYPE

 - Feature Pull Request

##### SUMMARY

In the notifier, we send events by creating a new connection to the communication medium on every new event. We can reduce the number of connections by sending messages via the Bot interface.

Fixes #270

### CODE REVIEW

**Comment by PrasadG193** [comment]
@codenio couldn't add you as a reviewer. PTAL

File: pkg/notify/slack.go [hunk 1]
```diff
@@ -98,12 +92,12 @@ func (s *Slack) SendEvent(event events.Event) error {
 func (s *Slack) SendMessage(msg string) error {
 	log.Debug(fmt.Sprintf(">> Sending to slack: %+v", msg))
 
-	api := slack.New(s.Token)
-	if len(s.SlackURL) != 0 {
-		api = slack.New(s.Token, slack.OptionAPIURL(s.SlackURL))
-	}
+	//api := slack.New(s.Token)
```
Comment (sanketsudake) [inline pkg/notify/slack.go:95]: please remove commented code.

File: pkg/notify/slack.go [hunk 2]
```diff
@@ -98,12 +92,12 @@ func (s *Slack) SendEvent(event events.Event) error {
...
+	//if len(s.SlackURL) != 0 {
+	//	api = slack.New(s.Token, slack.OptionAPIURL(s.SlackURL))
+	//}
```
Comment (PrasadG193) [inline pkg/notify/slack.go:98]:
```suggestion
```

**Comment by PrasadG193** [comment]
Sorry, the title was misleading. We are still creating a new connection each time.

**Pull request merged**